bionemo-recipes/recipes/esm2_accelerate_te/callbacks.py

```diff
@@ -32,3 +34,22 @@ def on_step_end(self, args: TrainingArguments, state: TrainerState, control: Tra
         """Interrupt training after a specified number of steps."""
         if state.global_step >= self.max_steps:
             control.should_training_stop = True
+
+
+class StepTimingCallback(TrainerCallback):
+    """Callback to log the time taken for each step."""
+
+    def __init__(self):
+        """Initialize the callback."""
+        self.step_start_time = None
+
+    def on_step_begin(self, args, state, control, **kwargs):
+        """Called at the beginning of each training step."""
+        self.step_start_time = time.perf_counter()
```

Comment:
would it make sense to keep a queue of maybe the 100 previous steps? and report `train/step_time` as `step_times[-1] - step_times[-2]`? And maybe also some averaged step time of `(step_times[-1] - step_times[0]) / len(step_times)`?